diff --git a/.gitignore b/.gitignore
index db6f1b9..f7fdcd9 100644
--- a/.gitignore
+++ b/.gitignore
@@ -4,6 +4,8 @@
 *.la
 *.so
 *.so.*
+*.wasm
+*.dylib
 gif2raw
 gif2rgb
 gifasm
diff --git a/Makefile b/Makefile
index f347162..17b3b25 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ DOCDIR = $(PREFIX)/share/doc/giflib
 CC ?= gcc
 OFLAGS = -g -fno-inline #-fsanitize=address
 OFLAGS  = -O2
-CFLAGS  += -std=gnu99 -fPIC -Wall $(OFLAGS)
+CFLAGS  += -std=gnu99 -fPIC -Wall -s USE_ZLIB=1 $(OFLAGS)
 #LDFLAGS += -fsanitize=address
 
 SHELL = /bin/sh
@@ -125,7 +125,7 @@ $(LIBGIFSO): $(OBJECTS) $(HEADERS)
 ifeq ($(UNAME), Darwin)
 	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(OBJECTS) -o $(LIBGIFSO)
 else
-	$(CC) $(CFLAGS) $(CPPFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBGIFSOMAJOR) -o $(LIBGIFSO) $(OBJECTS)
+	$(CC) $(CFLAGS) -s SIDE_MODULE=1 $(CPPFLAGS) -o $(LIBGIFSO) $(OBJECTS)
 endif
 
 libgif.a: $(OBJECTS) $(HEADERS)
@@ -135,14 +135,14 @@ $(LIBUTILSO): $(UOBJECTS) $(UHEADERS)
 ifeq ($(UNAME), Darwin)
 	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(UOBJECTS) -o $(LIBUTILSO)
 else
-	$(CC) $(CFLAGS) $(CPPLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILSOMAJOR) -o $(LIBUTILSO) $(UOBJECTS)
+	$(CC) $(CFLAGS) -s SIDE_MODULE=1 $(CPPLAGS) -o $(LIBUTILSO) $(UOBJECTS)
 endif
 
 libutil.a: $(UOBJECTS) $(UHEADERS)
 	$(AR) rcs libutil.a $(UOBJECTS)
 
 clean:
-	rm -f $(UTILS) $(OBSOLETE_UTILS) $(TARGET) libgetarg.a $(SHARED_LIBS) $(STATIC_LIBS) *.o
+	rm -f $(UTILS) $(OBSOLETE_UTILS) $(TARGET) libgetarg.a $(SHARED_LIBS) $(STATIC_LIBS) *.o *.wasm
 	rm -f $(LIBGIFSOVER)
 	rm -f $(LIBGIFSOMAJOR)
 	$(MAKE) --quiet -C doc clean
