diff --git a/.gitignore b/.gitignore
index f3f4abe..d86b687 100644
--- a/.gitignore
+++ b/.gitignore
@@ -4,6 +4,8 @@
 *.la
 *.so
 *.so.*
+*.wasm
+*.dylib
 gif2raw
 gif2rgb
 gifasm
diff --git a/Makefile b/Makefile
index 8497442..c939193 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@
 CC ?= gcc
 OFLAGS = -O0 -g
 OFLAGS  = -O2
-CFLAGS  += -std=gnu99 -fPIC -Wall -Wno-format-truncation $(OFLAGS)
+CFLAGS  += -std=gnu99 -fPIC -Wall -s USE_ZLIB=1 -Wno-format-truncation $(OFLAGS)
 
 SHELL = /bin/sh
 TAR = tar
@@ -36,7 +36,7 @@ SOURCES = dgif_lib.c egif_lib.c gifalloc.c gif_err.c gif_font.c \
 HEADERS = gif_hash.h  gif_lib.h  gif_lib_private.h
 OBJECTS = $(SOURCES:.c=.o)
 
-USOURCES = qprintf.c quantize.c getarg.c 
+USOURCES = qprintf.c quantize.c getarg.c
 UHEADERS = getarg.h
 UOBJECTS = $(USOURCES:.c=.o)
 
@@ -123,7 +123,7 @@ libutil.a: $(UOBJECTS) $(UHEADERS)
 	$(AR) rcs libutil.a $(UOBJECTS)
 
 clean:
-	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a $(LIBGIFSO) libutil.a $(LIBUTILSO) *.o
+	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a $(LIBGIFSO) libutil.a $(LIBUTILSO) *.o *.wasm
 	rm -f $(LIBGIFSOVER)
 	rm -f $(LIBGIFSOMAJOR)
 	$(MAKE) --quiet -C doc clean
